### PR TITLE
CPP2-17 Bluetooth communication

### PR DESCRIPTION
**Pull request by LeinadSokram**
Pls review thx :)

### CODE REVIEW

**Comment by LeinadSokram** [comment]
Had to solve a merge conflict, pls check if it's correct :)

File: NUCLEO-144_app/Src/main.c [hunk 1]
```diff
@@ -1313,24 +1313,54 @@ static void MX_GPIO_Init(void)
...
-            HAL_UART_Transmit(&huart3, (uint8_t *)"I turned off all 3 LEDs\r\n", 25,0xFFFF);
+        if (tmp[0] == '\n') {
+
+            buffer[counter - 1] = '\0';
+            counter = 0;
+
+            if (!strcmp((char *) buffer, "bt-on")) {
+                HAL_GPIO_WritePin(GPIOB, LD1_Pin, GPIO_PIN_SET);
+                HAL_UART_Transmit(&huart6,
+                        (uint8_t *) "I turned the bluetooth on!\r\n", 26, 100);
```
Comment (TheH3ROo) [inline NUCLEO-144_app/Src/main.c]: Why does it say"I turned the bluetooth on!" ? "HAL_GPIO_WritePin(GPIOB, LD1_Pin, GPIO_PIN_SET);" means you turn the green_led on. It has nothing to do with bluetooth.

File: NUCLEO-144_app/Src/main.c [hunk 2]
```diff
@@ -1313,24 +1313,54 @@ static void MX_GPIO_Init(void)
...
+            } else if (!strcmp((char *) buffer, "bt-off")) {
+                HAL_GPIO_WritePin(GPIOB, LD1_Pin, GPIO_PIN_RESET);
+                HAL_UART_Transmit(&huart6,
+                        (uint8_t *) "I turned the bluetooth off!\r\n", 27, 100);
```
Comment (TheH3ROo) [inline NUCLEO-144_app/Src/main.c]: Same with "... bluetooth on!", you just turn off the green_led here.

File: NUCLEO-144_app/Src/main.c [hunk 3]
```diff
@@ -1398,6 +1430,38 @@ void StartDefaultTask(void const * argument)
...
+    osDelay(1000);
+    HAL_UART_Transmit(&huart3, (uint8_t *) "FULLjobb\r\n", 10, 100);
+    htim1.Instance->CCR1 = 945;
+    osDelay(1000);
+    HAL_UART_Transmit(&huart3, (uint8_t *) "KOZEP\r\n", 7, 100);
+    htim1.Instance->CCR1 = 935;
+    osDelay(1000);
+
+    //for BT Communication
+    HAL_GPIO_WritePin(GPIOG, BT_nRESET_Pin, GPIO_PIN_SET);
```
Comment (TheH3ROo) [inline NUCLEO-144_app/Src/main.c]: This is the part where you actually turn on the bluetooth!

File: NUCLEO-144_app/Src/main.c [hunk 4]
```diff
@@ -1398,6 +1430,38 @@ void StartDefaultTask(void const * argument)
   /* Infinite loop */
   for(;;)
   {
+
+    HAL_UART_Transmit(&huart3, (uint8_t *)"1 sec elapsed\r\n", 15,0xFFFF);
+    osDelay(1000);
+
+    HAL_UART_Transmit(&huart3, (uint8_t *) "bal\r\n", 5, 100);
```
Comment (johnyka) [inline NUCLEO-144_app/Src/main.c]: I don't really get why this servo related code is showed as new code here. It was not your task to do it

File: NUCLEO-144_app/Src/main.c (hunk 1 above)
Comment (LeinadSokram) [reply NUCLEO-144_app/Src/main.c]: Just wanted to test if BT works by sending commands through Hercules. It should return a text message (besides turning the LED on). Sorry for the misleading message; next time I'll use proper prompts. ;)

File: NUCLEO-144_app/Src/main.c (hunk 2 above)
Comment (LeinadSokram) [reply NUCLEO-144_app/Src/main.c]: Same as before.

File: NUCLEO-144_app/Src/main.c (hunk 3 above)
Comment (LeinadSokram) [reply NUCLEO-144_app/Src/main.c]: Thanks for the tip ('adom')! Moved the code to the suggested line.

File: NUCLEO-144_app/Src/main.c (hunk 4 above)
Comment (LeinadSokram) [reply NUCLEO-144_app/Src/main.c]: Removed.

**Pull request merged**